core/src/container.cpp

```diff
@@ -825,6 +825,8 @@ void MergerPrivate::onNewPropagateSolution(const SolutionBase& s) {
 		ROS_ERROR_NAMED("Merger", "Only simple trajectories are supported");
 		return;
 	}
+	if (s.isFailure() || !trajectory->trajectory())  // ignore failure solutions
+		return;
```

Comment:
It's quite frustrating that Github forbids suggestions that include collapsed context lines...

If the container really can't work with empty trajectories, could you add `!trajectory->trajectory()` as an alternative condition to the check right above? If the merge receives a SubTrajectory without a real trajectory this should trigger an error and not be discarded silently.